### DISCUSSION TITLE
Road map 2022

### DISCUSSION THREAD

**Discussion by sanchezzzhak** [sanchezzzhak/node-device-detector#91] [General]
Version 3
* [ ] Migrate to typescript
* [ ] Add support bun runner https://github.com/oven-sh/bun
* [ ] think about low-level optimizations on rust lang

**Comment by L0ngB0n3pUn3sc00n669699** [comment]
I need to know the best way to reset forgotten pattern on Motorola moto e XT2052DL  without looking data???

**Reply by sanchezzzhak** [reply]
you need to contact the specialized sites
* https://forum.xda-developers.com
* https://4pda.to/forum/

as a standard, this is solved by restoring access through google account.

otherwise, you will have to reset the phone to factory settings
